src/test/java/com/qa/koel/BaseTest.java

```diff
@@ -81,4 +79,22 @@ public String getSuccessMessage(){
         return notification.getText();
 
     }
+    public void choosePlaylistByName (String playlistName) {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(text(),'" + playlistName + "')]"))).click();
+    }
+
+    public int countSongs() {
+        return driver.findElements(By.cssSelector("section#playlistWrapper td.title")).size();
+    }
+    public String getPlaylistDetails() {
+        return driver.findElement(By.cssSelector("span.meta.text-secondary span.meta")).getText();
+    }
+    public void displayAllSongs() {
+        List<WebElement> songList = driver.findElements(By.cssSelector("section#playlistWrapper td.title"));
+        System.out.println("Number of songs found "+countSongs());
+        for (WebElement e : songList) {
+            System.out.println(e.getText());
+        }
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.

**BaseTest** — Implements the necessary methods for work with webdriver.

The BaseTest class contains common setup and teardown methods, as well as any shared functionality or configurations required by multiple test classes.

**BaseTest**

The problem we are trying to solve here is the reuse of common behaviors across test classes avoiding coding duplication and centralizing these actions at one point.

**@BeforeSuite**	- The annotated method will be run before all tests in this suite have run.

**@BeforeMethod** -	The annotated method will be run before all the test methods in the current class have been run.

**@AfterMethod**	- The annotated method will be run after each test method.

_Example without a Base Test Class_

![image](https://github.com/Test-Pro-Bootcamp/A53-QA-Automation-Framework/assets/104761482/c3abad8a-8bc1-4ad0-8a66-23bd19e7af48)